messagequeue/sqs/validation.go

```diff
@@ -17,11 +18,12 @@ const (
 	period            = "."
 )
 
+var nameAllowedCharacters = regexp.MustCompile(`^[a-zA-Z0-9_\-\.]+$`)
 var (
 	prohibitedPrefixError = errors.
 				New("name has invalid prefix (%s|%s)", prohibitedAmazon, prohibitedAWS)
 	dotError         = errors.New("name cannot begin, end, or contain sequences of '%s'", period)
-	bodyMinimumError = "minimum character count is 1"
+	bodyMinimumError = errors.New("minimum character count is 1")
```

Comment:
we don't want to initialize the error here because then the stack trace is static and will not be useful for whoever logs it.